samples/core/parameterized_tfx_oss/taxi_pipeline_notebook.ipynb

```diff
@@ -0,0 +1,343 @@
+    "learning platform based on TensorFlow. It provides a configuration framework to express ML pipelines\n",
+    "consisting of TFX components, which brings the user large-scale ML task orchestration, artifact lineage, as well as the power of various [TFX libraries](https://www.tensorflow.org/resources/libraries-extensions). Kubeflow Pipelines can be used as the orchestrator supporting the \n",
+    "execution of a TFX pipeline.\n",
+    "\n",
+    "This sample demonstrates how to author a ML pipeline in TFX and run it on a KFP deployment. \n",
+    "\n",
+    "## Permission\n",
+    "\n",
+    "This pipeline requires Google Cloud Storage permission to run. \n",
+    "If KFP was deployed through K8S marketplace, please follow instructions in [the guideline](https://github.com/kubeflow/pipelines/blob/master/manifests/gcp_marketplace/guide.md#gcp-service-account-credentials)\n",
+    "to make sure the service account has `storage.admin` role."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python3 -m pip install pip --upgrade --quiet --user\n",
+    "!python3 -m pip install kfp --upgrade --quiet --user\n",
+    "!python3 -m pip install tfx --upgrade --quiet --user"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example we'll need a very recent version of TFX SDK to leverage the [`RuntimeParameter`](https://github.com/tensorflow/tfx/blob/93ea0b4eda5a6000a07a1e93d93a26441094b6f5/tfx/orchestration/data_types.py#L137) feature.\n",
+    "\n",
+    "## RuntimeParameter in TFX DSL\n",
+    "Currently, TFX DSL only supports parameterizing field in the `PARAMETERS` section of `ComponentSpec`, see [here](https://github.com/tensorflow/tfx/blob/93ea0b4eda5a6000a07a1e93d93a26441094b6f5/tfx/types/component_spec.py#L126). This prevents runtime-parameterizing the pipeline topology. Also, if the declared type of the field is a protobuf, the user needs to pass in a dictionary with exactly the same names for each field, and specify one or more value as `RuntimeParameter` objects. In other word, the dictionary should be able to be passed in to [`ParseDict()` method](https://github.com/protocolbuffers/protobuf/blob/04a11fc91668884d1793bff2a0f72ee6ce4f5edd/python/google/protobuf/json_format.py#L433) and produce the correct pb message."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python3 -m pip install --quiet --index-url https://test.pypi.org/simple/ tfx==0.16.0.dev20191212 --user"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from typing import Optional, Text\n",
+    "\n",
+    "import kfp\n",
+    "from kfp import dsl\n",
+    "\n",
+    "from tfx.components.evaluator.component import Evaluator\n",
```

Comment:
JFYI: You can import TFX components like this: `from tfx.components import Evaluator`

Reply:
Indeed. Forgot that shortcut :)